src/Lean/Server/FileWorker.lean

```diff
@@ -273,21 +273,22 @@ section Updates
   def updateDocument (newMeta : DocumentMeta) : WorkerM Unit := do
     let ctx ← read
     let oldDoc := (←get).doc
-    let ⟨cmdSnaps, e?⟩ ← oldDoc.cmdSnaps.updateFinishedPrefix
-    match e? with
-    | some ElabTaskError.aborted =>
-      -- This case should not be possible. only the main task aborts tasks and ensures that aborted tasks
-      -- do not show up in `snapshots` of an EditableDocument.
-      throwServerError "Internal server error: elab task was aborted while still in use."
-    | some (ElabTaskError.ioError ioError) => throw ioError
-    | _ => -- No error or EOF
-      -- The watchdog only restarts the file worker when the semantic content of the header changes.
-      -- If e.g. a newline is deleted, it will not restart this file worker, but we still
-      -- need to reparse the header so that the offsets are correct.
-      let (newHeaderStx, newMpState, _) ← Parser.parseHeader newMeta.mkInputContext
-      let newHeaderSnap := { cmdSnaps.finishedPrefix.head! with stx := newHeaderStx, mpState := newMpState }
+    -- The watchdog only restarts the file worker when the semantic content of the header changes.
+    -- If e.g. a newline is deleted, it will not restart this file worker, but we still
+    -- need to reparse the header so that the offsets are correct.
+    let (newHeaderStx, newMpState, _) ← Parser.parseHeader newMeta.mkInputContext
+    let cancelTk ← CancelToken.new
+    -- Wait for at least one snapshot from the old doc, we don't want to unnecessarily re-run `print-paths`
+    let headSnapTask ← oldDoc.cmdSnaps.waitHead?
+    let newSnaps ← EIO.mapTask (ε := ElabTaskError) (t := headSnapTask) fun headSnap?? => do
+      let headSnap? ← MonadExcept.ofExcept headSnap??
+      -- There is always at least one snapshot absent exceptions
+      let headSnap := headSnap?.get!
+      let newHeaderSnap := { headSnap with stx := newHeaderStx, mpState := newMpState }
       oldDoc.cancelTk.set
       let changePos := oldDoc.meta.text.source.firstDiffPos newMeta.text.source
+      -- Ignore exceptions, we are only interested in the successful snapshots
+      let (cmdSnaps, _) ← oldDoc.cmdSnaps.updateFinishedPrefix
```

Comment:
A coincidental simplification, but it seems reasonable to ignore errors from the old document. This subsumes the "Internal server error" from above.

Reply:
Should we still have the check for `aborted` here? It's basically just an assertion to make sure we don't violate assumptions internally.

Reply:
I don't know, does the assertion have any bearing on the surrounding code? As the comment says, all we're here for is recycling snapshots before the old document gets dumped.

Reply:
No, it is a "non-local" check which verifies that a totally different part of the code behaves correctly. Maybe this isn't great practice so makes sense to leave out.